src/store/ChartAdapterStore.ts

```diff
@@ -237,38 +241,58 @@ export default class ChartAdapterStore {
     }
 
     onTouch(e: TouchEvent) {
-        const chartNode = this.mainStore.chart.chartNode;
         // Prevent vertical scroll on the chart for touch devices by forcing scroll on a scrollable parent of the chart:
+        const chartNode = this.mainStore.chart.chartNode;
         if (
             chartNode &&
             this.scrollableChartParent &&
             !this.mainStore.state.isVerticalScrollEnabled &&
-            e.touches.length === 1
+            e.changedTouches.length === 1
         ) {
-            const { pageX, screenX, screenY } = e.touches[0];
-            if (['touchstart', 'touchend'].includes(e.type)) {
-                this.touchValues = e.type === 'touchstart' ? { x: screenX, y: screenY } : { yOnTouchEnd: screenY };
-            } else if (e.type === 'touchmove') {
+            const { pageX, pageY } = e.changedTouches[0];
+
+            if (['touchmove', 'touchend'].includes(e.type)) {
                 const nonScrollableAreaWidth = chartNode.offsetWidth - this.mainStore.chart.yAxisWidth;
                 const { left } = chartNode.getBoundingClientRect();
 
                 if (this.touchValues.x && this.touchValues.y) {
-                    const deltaX = Math.abs(screenX - this.touchValues.x);
-                    const deltaY = Math.abs(screenY - this.touchValues.y);
+                    const deltaX = Math.abs(pageX - this.touchValues.x);
+                    const deltaY = Math.abs(pageY - this.touchValues.y);
+                    this.touchValues.deltaYTotal = (this.touchValues.deltaYTotal ?? 0) + (this.touchValues.y - pageY);
                     const isVerticalScroll = deltaY > deltaX;
                     const x = pageX - left;
-                    if (x < nonScrollableAreaWidth && isVerticalScroll && !this.scrollChartParentOnTouchTimer) {
-                        this.touchValues.yOnTouchEnd = undefined;
-                        this.scrollChartParentOnTouchTimer = setTimeout(() => {
-                            this.scrollableChartParent?.scrollBy({
-                                top: screenY - Number(this.touchValues.yOnTouchEnd ?? this.touchValues.y),
+                    const isForcedScrollArea = x < nonScrollableAreaWidth;
+                    const shouldForceMaxScroll =
+                        Math.abs(Number(this.touchValues.deltaYTotal)) > 10 && e.type === 'touchend';
```

Comment:
Sometimes we get `touchcancel` instead of `touchend`, do we need to handle this case too?

Reply:
no, we won't get `touchcancel` here ever because we don't trigger `onTouch` action for this event type.